src/patch_Distributions.jl

```diff
@@ -0,0 +1,82 @@
+### Maxwell Boltzmann distribution
+# https://en.wikipedia.org/wiki/Maxwell–Boltzmann_distribution
+
+const sqrt_two_over_pi = sqrt(2 / pi)
+const _CHI3 = Distributions.Chi(3; check_args=false)
+
+struct MaxwellBoltzmann{T<:Real} <: Distributions.ContinuousUnivariateDistribution
+    a::T # scale
+    MaxwellBoltzmann{T}(a::T) where {T} = new{T}(a)
+end
+
+function MaxwellBoltzmann(a::T; check_args::Bool=true) where {T<:Real}
+    Distributions.@check_args MaxwellBoltzmann (a, a > zero(a))
+    return MaxwellBoltzmann{T}(a)
+end
+
+function MaxwellBoltzmann(a::Integer; check_args::Bool=true)
+    return MaxwellBoltzmann(float(a); check_args=check_args)
+end
+MaxwellBoltzmann() = MaxwellBoltzmann(1.0)
+
+Distributions.@distr_support MaxwellBoltzmann 0.0 Inf
+
+### Conversions
+convert(::Type{MaxwellBoltzmann{T}}, a::S) where {T<:Real,S<:Real} = MaxwellBoltzmann(T(a))
+function Base.convert(::Type{MaxwellBoltzmann{T}}, d::MaxwellBoltzmann) where {T<:Real}
+    return MaxwellBoltzmann(T(d.a))
+end
+Base.convert(::Type{MaxwellBoltzmann{T}}, d::MaxwellBoltzmann{T}) where {T<:Real} = d
+
+### Parameters
+Distributions.scale(d::MaxwellBoltzmann) = d.a
+Distributions.params(d::MaxwellBoltzmann) = (d.a,)
+Distributions.partype(d::MaxwellBoltzmann{T}) where {T} = T
+
+### Statistics
+Distributions.mean(d::MaxwellBoltzmann) = 2 * sqrt_two_over_pi * d.a
+Distributions.median(d::MaxwellBoltzmann{T}) where {T<:Real} = d.a * T(1.5381722544550522) # a * sqrt(2 Q^(-1)(3/2, 1/2))
```

Comment:
Is there an advantage to using a magic number with a comment instead of just putting in the formula directly?

Reply:
The calculation of the magic number involves computing a special function on some constants. Beyond avoiding single-call dependencies (`SpecialFunctions.jl` in this case), it is rather slow to call these types of functions (the inverse regularized gamma function `Q^-1` in this case): 

```Julia
using QEDevents
using Distributions
using SpecialFunctions
using BenchmarkTools

median_(d) = d.a*sqrt(2*gamma_inc_inv(3/2,1/2,1/2))

d=Maxwell(0.5)
```
Then we have the following Benchmark results:

```Julia
julia> @benchmark median($d)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  3.531 ns … 6.223 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     3.541 ns             ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.543 ns ± 0.062 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

      ▁  ▇ ▆ ▆  ▇ ▆ ▅  ▇ ▇  ▆ ▅ █  ▅ ▅ ▇  ▇ ▁
  ▂▁▃▁█▁▁█▁█▁█▁▁█▁█▁█▁▁█▁█▁▁█▁█▁█▁▁█▁█▁█▁▁█▁█▁▁▆▁▃▁▂▁▁▂▁▂▁▂ ▄
  3.53 ns        Histogram: frequency by time       3.56 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```

```Julia 
julia> @benchmark median_($d)
BenchmarkTools.Trial: 10000 samples with 151 evaluations.
 Range (min … max):  679.483 ns …   8.604 μs  ┊ GC (min … max): 0.00% … 89.35%
 Time  (median):     690.550 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   706.220 ns ± 273.602 ns  ┊ GC (mean ± σ):  1.59% ±  3.72%

      ▁▁▃▅▇██▇▇▆▆▅▅▄▃▂▂▂▃▃▂▁                     ▁▁▂▁▁▁         ▂
  ▄▅▆████████████████████████▇▆▆▇▆▅▇▆▆▇▆▆▆▅▅▅▇▆▅▇████████▇▆▄▆▅▄ █
  679 ns        Histogram: log(frequency) by time        738 ns <

 Memory estimate: 608 bytes, allocs estimate: 2.
```

Reply:
I would have expected Julia to realize that this is just a constant and compile it away...

Reply:
Usually, Julia can compile it away, but unfortunately, the special function is the obstacle in this case.

Reply:
That's unfortunate then, but okay :+1: